core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/core/validation/ValidateRequest.java

```diff
@@ -254,11 +264,19 @@ void setRuleParameters(Map<String, Object> ruleParameters) {
 	private static final class IssuesToPersist {
 		
 		public final String ruleId;
-		public final Collection<ComponentIdentifier> affectedComponentIds;
-
-		public IssuesToPersist(String ruleId, Collection<ComponentIdentifier> affectedComponentIds) {
+		public Collection<ValidationIssueDetails> issueDetails;
+		
+		@SuppressWarnings("unchecked")
+		public <T> IssuesToPersist(String ruleId, Collection<T> evaluationResult) {
 			this.ruleId = ruleId;
-			this.affectedComponentIds = affectedComponentIds;
+			if (evaluationResult.iterator().hasNext() && evaluationResult.iterator().next() instanceof ValidationIssueDetails) {
+				this.issueDetails = (Collection<ValidationIssueDetails>) evaluationResult;
+			} else {
+				this.issueDetails = evaluationResult.parallelStream()
```

Comment:
You should not use `parallelStream` to process a huge list like this, none of the downstream consumers do anything that would indicate the necessity of parallel execution.